Description: Check if binary files for previewing Gtk themes are available
 This patch is necessary for separating a package with dependency from Gtk libs.
Author: Boris Pek <tehnick-8@mail.ru>
Last-Update: 2012-11-06

--- a/src/gtkconfigkcmodule.cpp
+++ b/src/gtkconfigkcmodule.cpp
@@ -78,13 +78,22 @@
     ui->gtk2Preview->setIcon(KIcon("document-preview")); //!! for some reason it doesn't work with QIcon::fromTheme
     ui->gtk3Preview->setIcon(KIcon("document-preview"));
     
+    ui->gtk2Preview->hide();
+    ui->gtk3Preview->hide();
+    
     m_p2 = new KProcess(this);
     m_p2->setEnv("GTK2_RC_FILES", m_tempGtk2Preview, true);
-    *m_p2 << KStandardDirs::findExe("gtk_preview");
+    if (KStandardDirs::findExe("gtk_preview") != QString::null) {
+        *m_p2 << KStandardDirs::findExe("gtk_preview");
+        ui->gtk2Preview->show();
+    }
     
     m_p3 = new KProcess(this);
     m_p3->setEnv("XDG_CONFIG_HOME", KGlobal::dirs()->saveLocation("tmp", ".config"));
-    *m_p3 << KStandardDirs::findExe("gtk3_preview");
+    if (KStandardDirs::findExe("gtk3_preview") != QString::null) {
+        *m_p3 << KStandardDirs::findExe("gtk3_preview");
+        ui->gtk3Preview->show();
+    }
     
     //UI changes
     connect(ui->cb_theme, SIGNAL(currentIndexChanged(int)), this, SLOT(appChanged()));
